### PR TITLE
feat: add dictionary get and set commands

### PR DESCRIPTION
**Pull request by tylerburdsall**
PR for https://github.com/momentohq/client-sdk-rust/pull/75 so that automated checks pass.

I checked this out locally and used a test token, everything passed.

Commit log:

- feat: add dictionary get and set commands
- fixup doc comment
- fix doctest
- working through clippy lints
- fix more clippy lint failures
- fix more clippy lint failures
- more lint fix
- fix: compile error in docs test

### CODE REVIEW

File: src/simple_cache_client.rs [hunk 1]
```diff
@@ -497,6 +501,181 @@ impl SimpleCacheClient {
...
+                    dictionary: Some(dictionary),
+                })
+            }
+            Some(Dictionary::Missing(_)) | None => {
+                // unclear if `None` variant here should be treated same as missing
+                Ok(MomentoDictionaryGetResponse {
+                    result: MomentoDictionaryGetStatus::MISSING,
+                    dictionary: None,
+                })
+            }
```
Comment (tylerburdsall) [inline src/simple_cache_client.rs:675]: @malandis I believe you have the answer to this one

File: src/simple_cache_client.rs [hunk 2]
```diff
@@ -497,6 +501,181 @@ impl SimpleCacheClient {
...
+                let mut dictionary = HashMap::new();
+
+                // map the dictionary response parts to get responses
+                for (field, item) in fields.drain(..).zip(response.items.drain(..)) {
+                    match item.result() {
+                        ECacheResult::Hit => {
+                            dictionary.insert(field, item.cache_body);
+                        }
+                        ECacheResult::Miss => {}
+                        _ => todo!(),
```
Comment (tylerburdsall) [inline src/simple_cache_client.rs:660]: @malandis what was the result we wanted here?

File: src/simple_cache_client.rs (hunk 1 above)
Comment (tylerburdsall) [reply src/simple_cache_client.rs:675]: Oh wait, Michael is OOO, @cprice404 would you happen to know this?

File: src/simple_cache_client.rs (hunk 2 above)
Comment (tylerburdsall) [reply src/simple_cache_client.rs:660]: Same thing here @cprice404

File: src/simple_cache_client.rs (hunk 1 above)
Comment (cprice404) [reply src/simple_cache_client.rs:675]: based on this: 

https://github.com/momentohq/client-sdk-dotnet-incubating/blob/main/src/Momento.Sdk.Incubating/Internal/ScsDataClient.cs#L315


I believe that we are always returning `Miss`, unless the wire response was `Found`.

File: src/simple_cache_client.rs (hunk 2 above)
Comment (cprice404) [reply src/simple_cache_client.rs:660]: I think this is the relevant code from .NET:

https://github.com/momentohq/client-sdk-dotnet-incubating/blob/main/src/Momento.Sdk.Incubating/Internal/ScsDataClient.cs#L315

File: clippy.toml [hunk 3]
```diff
@@ -0,0 +1 @@
+allow-expect-in-tests = true
```
Comment (brayniac) [inline clippy.toml:1]: Do we need the `#[allow::clippy(unwrap_used)]` now that this is set?

Comment (tylerburdsall) [reply clippy.toml:1]: Oh likely not now that we've got this set! I have some more commits to push, I'll include this

File: src/simple_cache_client.rs [hunk 4]
```diff
@@ -497,6 +501,181 @@ impl SimpleCacheClient {
...
+    /// *NOTE*: This is preview functionality and requires that you contact
+    /// Momento Support to enable these APIs for your cache.
+    ///
+    /// # Arguments
+    ///
+    /// * `cache_name` - name of cache
+    /// * `dictionary_name` - name of the dictionary
+    /// * `dictionary` - hashmap of dictionary key-value pairs
+    /// * `ttl_seconds` - If None is passed, uses the client's default ttl
+    /// * `refresh_ttl`
```
Comment (brayniac) [inline src/simple_cache_client.rs:515]: I wasn't sure of the exact behavior here. Would be nice to note what true/false mean for this.

Comment (tylerburdsall) [reply src/simple_cache_client.rs:515]: Great feedback, I can document what this means

Comment (tylerburdsall) [reply src/simple_cache_client.rs]: Addressed in https://github.com/momentohq/client-sdk-rust/pull/76/commits/362fa94c42740de361132cf7371b0a912ca2cdd1

File: src/simple_cache_client.rs (hunk 2 above)
Comment (tylerburdsall) [reply src/simple_cache_client.rs]: Thanks Chris! This has been addressed

File: src/simple_cache_client.rs (hunk 1 above)
Comment (tylerburdsall) [reply src/simple_cache_client.rs]: Perfect, I've addressed this!

File: src/endpoint_resolver.rs [hunk 5]
```diff
@@ -116,6 +116,7 @@ impl MomentoEndpointsResolver {
 }
 
 #[cfg(test)]
+#[allow(clippy::unwrap_used)]
```
Comment (brayniac) [inline src/endpoint_resolver.rs:119]: I think this can be removed now.

File: src/jwt.rs [hunk 6]
```diff
@@ -50,6 +50,7 @@ fn token_parsing_error() -> MomentoError {
 }
 
 #[cfg(test)]
+#[allow(clippy::unwrap_used)]
```
Comment (brayniac) [inline src/jwt.rs:53]: Same here.

**Review by brayniac — APPROVED** [review]
Ignoring the minor cleanups of my module level clippy allows, this LGTM. Thanks for all the help getting this ready @tylerburdsall

File: src/endpoint_resolver.rs (hunk 5 above)
Comment (tylerburdsall) [reply src/endpoint_resolver.rs]: Thanks, missed this one! Addressed in https://github.com/momentohq/client-sdk-rust/pull/76/commits/5a8b158f5f3cc26c7a04aedb4f5c9cb6d51e387e

File: src/jwt.rs (hunk 6 above)
Comment (tylerburdsall) [reply src/jwt.rs]: Same thing, addressed in https://github.com/momentohq/client-sdk-rust/commit/5a8b158f5f3cc26c7a04aedb4f5c9cb6d51e387e

**Review by brayniac — APPROVED** [review]
LGTM! Thanks!

**Pull request merged**